provider/testing/resource.go

```diff
@@ -23,19 +23,19 @@ import (
 )
 
 type ResourceTestCase struct {
-	Provider       *provider.Provider
-	Table          *schema.Table
-	Config         string
-	SkipEmptyJsonB bool
-	// SkipEmptyColumn will skip checking results for empty columns
-	SkipEmptyColumn bool
-	// SkipEmptyRows will skip checking that results were returned and will just check that fetch worked
-	SkipEmptyRows bool
+	Provider *provider.Provider
+	Config   string
+	// we want it to be parallel by default
+	NotParallel bool
+	// ParallelFetchingLimit limits parallel resources fetch at a time
+	ParallelFetchingLimit uint64
 }
 
 // IntegrationTest - creates resources using terraform, fetches them to db and compares with expected values
 func TestResource(t *testing.T, resource ResourceTestCase) {
-	t.Parallel()
+	if !resource.NotParallel {
+		t.Parallel()
+	}
```

Comment:
Why do you need `NotParallel` why not just check if `ParallelFetchingLimit > 1` ?

Reply:
Those are two different things `ParallelFetchingLimit` is just the max number of multiplexer cq-provider-sdk is spawning and `Parallel` means if the `go test` will run those tests in parallel or not. if for example we will run in parallel two integration tests for the same resources than it will raise pk errors.